### PR TITLE
Organize B2B topics & update info for few articles

### PR DESCRIPTION
**Pull request by sushmangupta**
- Rearranging the structure for B2B concepts as discussed in the meeting with @Haberkamp , @LarsKemper , @King-of-Babylon , @Isengo1989 . 
- Create permission via Apps under employee management updated
- Added "TODO" pointers to include more information from the dev team or respective PO.
- Added comments to bring clarity to the article
- Added new content submitted by the dev team

### CODE REVIEW

File: products/extensions/b2b/b2b-components/employee-management/guides/b2b-employee-invite-registration.md
Comment (Haberkamp) [inline products/extensions/b2b/b2b-components/employee-management/guides/b2b-employee-invite-registration.md:1]: We have this documented in our stoplight documentation? Is this enough? Then we do not need extra guide, which would just explain the same concept just in another way.

File: products/extensions/b2b/b2b-components/employee-management/guides/b2b-employee-invite-registration.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+# B2B Employee invite / registration
+
+//TODO more info required here
```
Comment (sushmangupta) [inline products/extensions/b2b/b2b-components/employee-management/guides/b2b-employee-invite-registration.md:3]:
```suggestion
Employees can be created via storefront and api to get an invite mail. This has to be later confirmed by the employee to set a password.
```

**Review by LarsKemper — APPROVED** [review]

File: products/extensions/b2b/README.md [hunk 2]
```diff
@@ -0,0 +1,3 @@
+# B2B Extension
+
+Unlike the B2B Suit, the B2B Components aim to provide a flexible set of B2B-related features. This enables B2B merchants to deliver the experience they want to their business partners. This is achieved by providing abilities that allow B2B Merchants to conduct basic B2B jobs and enable agencies to extend the Components according to the merchant's specific requirements.
```
Comment (King-of-Babylon) [inline products/extensions/b2b/README.md:3]:
```suggestion
Unlike the B2B Suite, the B2B Components aim to provide a flexible set of B2B-related features. This enables B2B merchants to deliver the experience they want to their business partners. This is achieved by providing abilities that allow B2B Merchants to conduct basic B2B jobs and enable agencies to extend the Components according to the merchant's specific requirements.
```

File: products/extensions/b2b/b2b-components/employee-management/concept/README.md [hunk 3]
```diff
@@ -0,0 +1,8 @@
+# Concept
+
+This section includes are the concepts related to Employee management.
```
Comment (King-of-Babylon) [inline products/extensions/b2b/b2b-components/employee-management/concept/README.md:3]:
```suggestion
This section includes are the concepts related to Employee Management.
```

File: products/extensions/b2b/b2b-components/employee-management/guides/README.md [hunk 4]
```diff
@@ -0,0 +1,3 @@
+# Guides
+
+The following articles gives you an idea of roles. Also it guides you on creating your own permissions via app or plugin for the b2b employee management component.
```
Comment (King-of-Babylon) [inline products/extensions/b2b/b2b-components/employee-management/guides/README.md:3]:
```suggestion
The following articles gives you an idea of roles. Also it guides you on creating your own permissions via app or plugin for the B2B Employee Management component.
```

File: products/extensions/b2b/b2b-components/employee-management/guides/b2b-permissions/README.md [hunk 5]
```diff
@@ -0,0 +1,20 @@
+# B2B permissions
+
+Use permissions to restrict access to certain information or functionalities within the B2B components. For example, the B2B supervisor can restrict which employee can manage the company's employee accounts.
```
Comment (King-of-Babylon) [inline products/extensions/b2b/b2b-components/employee-management/guides/b2b-permissions/README.md:3]:
```suggestion
Use permissions to restrict access to certain information or functionalities within the B2B Components. For example, the B2B supervisor can restrict which employee can manage the company's employee accounts.
```

File: products/extensions/b2b/b2b-components/employee-management/guides/b2b-permissions/README.md [hunk 6]
```diff
@@ -0,0 +1,20 @@
...
+
+## Groups
+Permissions are divided into individual groups that have a logical relationship to each other.
+
+## Dependencies
+A permission can be dependent on another permission, without which this permission cannot be used. For example, if a role is created with the permission to edit employee accounts, this role must also have the permission to view employee accounts. This is because the `employee.edit` permission depends on the `employee.read` permission.
+
+## Shopware base permissions
+The following permissions are already included and used in the B2B employee management component. More "base" permissions will be duly added with future B2B Components.
```
Comment (King-of-Babylon) [inline products/extensions/b2b/b2b-components/employee-management/guides/b2b-permissions/README.md:12]:
```suggestion
The following permissions are already included and used in the B2B Employee Management component. More "base" permissions will be duly added with future B2B Components.
```

File: products/extensions/b2b/b2b-components/employee-management/guides/b2b-roles.md [hunk 7]
```diff
@@ -0,0 +1,5 @@
+# B2B Roles
+
+Roles can be used to bind multiple permission to employees with contexts. Every employee can have one assigned role. Based on that role and the containing permission, the employee will get access to certain information and functionalities.
+
+The Business partner has the opportunity to create a “default” role that will be selected by default, when creating a new employee.
```
Comment (King-of-Babylon) [inline products/extensions/b2b/b2b-components/employee-management/guides/b2b-roles.md:5]:
```suggestion
The business partner has the opportunity to create a “default” role that will be selected by default, when creating a new employee.
```

**Review by King-of-Babylon — COMMENTED** [review]
Only a few upper and lower case comments :blush:

**Review by King-of-Babylon — APPROVED** [review]

**Review by bojanrajh — CHANGES_REQUESTED** [review]
Content looks good.

Please, add [redirects](https://github.com/shopware/docs/blob/master/.gitbook.yaml) for all the articles that were moved:
 - `products/extensions/b2b-suite/*: products/extensions/b2b/b2b-suite/*`
 - `...`

**Comment by LarsKemper** [comment]
> Content looks good.
> 
> Please, add [redirects](https://github.com/shopware/docs/blob/master/.gitbook.yaml) for all the articles that were moved:
> 
> * `products/extensions/b2b-suite/*: products/extensions/b2b/b2b-suite/*`
> * `...`

For the redirects doesn't work at all. Event the existing ones are redirecting me to the 404 page
For example:

**Comment by Isengo1989** [comment]
@LarsKemper I guess there was still old code/urls from gitbook, since we moved recently to devhub there was a change in the URL path endings, @bojanrajh fixed this recently.

https://github.com/shopware/docs/commit/732c446c6827cfcc7aa938b9967d8cd6d1fb2315

Those redirects work now. Could you take another look?

**Comment by King-of-Babylon** [comment]
> @LarsKemper I guess there was still old code/urls from gitbook, since we moved recently to devhub there was a change in the URL path endings, @bojanrajh fixed this recently.
> 
> [732c446](https://github.com/shopware/docs/commit/732c446c6827cfcc7aa938b9967d8cd6d1fb2315)
> 
> Those redirects work now. Could you take another look? If you need help, let me know (but you probably know better what was moved. If you pass me a list, I can look into it as well)

Hi @Isengo1989,

I guess the redirects should look like that:

- products/extensions/b2b-components/employee-management/concept/index.md: products/extensions/b2b/b2b-components/employee-management/concept/README.md

- products/extensions/b2b-components/employee-management/guides/creating-own-permissions-via-app.md: products/extensions/b2b/b2b-components/employee-management/guides/b2b-permissions/creating-own-permissions-via-app.md

- products/extensions/b2b-components/employee-management/guides/index.md: products/extensions/b2b/b2b-components/employee-management/guides/README.md

- products/extensions/b2b-components/employee-management/index.md: products/extensions/b2b/b2b-components/employee-management/README.md

- products/extensions/b2b-components/index.md: products/extensions/b2b/b2b-components/README.md

- products/extensions/b2b-suite/guides/administration/index.md: products/extensions/b2b/b2b-suite/guides/administration/README.md

**Comment by Isengo1989** [comment]
@bojanrajh I rebased it and adjusted the mentioned topics. Please check the redirects (question about wildcards - slack)

**Pull request merged**